mypy import-untyped error due to missing py.typed

I tried running mypy on project using fast-agent and it threw a import-typed error

```
error: Skipping analyzing "mcp_agent.core.fastagent": module is installed, but missing library stubs or py.typed marker  [import-untyped]
```

It seems that while the github repo includes a `py.typed` file, the packaged version when installing using uv does not. 

Proposing #305 to include the py.typed file explicitly in the package.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -75,6 +75,7 @@ include = [
     "src/mcp_agent/**/*.csv",
     "src/mcp_agent/**/*.txt",    
     "src/mcp_agent/**/*.example",
+    "src/mcp_agent/py.typed",
     "examples/**/*",
 ]
 

